vendor/github.com/Juniper/contrail-go-api/keystone.go

```diff
@@ -135,6 +153,34 @@ func (kClient *KeystoneClient) Authenticate() error {
 	return nil
 }
 
+func (kClient *KeepaliveKeystoneClient) needsRefreshing() bool {
+	if kClient.current == nil {
+		return true
+	}
+
+	issuedAtTime, err := time.Parse(time.RFC3339, kClient.current.Issued_At)
+	if err != nil {
+		return true
+	}
+
+	expires, err := time.Parse(time.RFC3339, kClient.current.Expires)
+	if err != nil {
+		return true
```

Comment:
I'm not sure I like this error handling. If anything, I think we should do something like
```
func (kClient *KeepaliveKeystoneClient) needsRefreshing() (bool, error) {
   ...
   issuedAtTime, err := time.Parse(time.RFC3339, kClient.current.Expires)
   if err != nil {
      return false, err
   }
  ...
  // (same for "expires")
}


func (kClient *KeepaliveKeystoneClient) AddAuthentication(req *http.Request) error {
   refresh, err := kClient.needsRefreshing()
   if err != nil { return err }
   if refresh { kClient.current = nil }
   return ....AddAuthentication(req)
}
```
What do you think?

Reply:
maybe other reviewers could comment on this as well? @PawelMarchewkaCodiLime @jablonskim